components/brave_rewards/resources/page/components/adsBox.tsx

```diff
@@ -392,6 +392,17 @@ class AdsBox extends React.Component<Props, State> {
     return false
   }
 
+  hasLikedEntries = (adHistory: Rewards.AdsHistory[]) => {
+    for (let ix = 0; ix < adHistory.length; ix++) {
```

Comment:
Not 100% sure, but looks like maybe this could be something like:

```js
for (const historyItem of adHistory) {
  for (const row of historyItem.adDetailRows) {
    if (row.adContent.likeAction === 1) {
      return true
    }
  }
}
return false
```

or maybe even:

```js
return adHistory.some((adHistoryItem) => {
  return adHistoryItem.adDetailRows.some((row) => {
    return row.adContent.likeAction === 1
  })
})
```

Reply:
Thanks @zenparsing! That looks much nicer. Updated via https://github.com/brave/brave-core/pull/13496/commits/a8dd17f6709b9eca1480226b68d9af8ddc90d730